crates/biome_cli/src/reporter/gitlab.rs

```diff
@@ -0,0 +1,243 @@
+use crate::{DiagnosticsPayload, Execution, Reporter, ReporterVisitor, TraversalSummary};
+use biome_console::{markup, Console, ConsoleExt};
+use biome_diagnostics::{LineIndexBuf, PrintDescription};
+use serde::Serialize;
+use std::{
+    cmp::max,
+    collections::HashSet,
+    fs::canonicalize,
+    hash::{DefaultHasher, Hash, Hasher},
+    path::{Path, PathBuf},
+};
+
+/// An entry in the GitLab Code Quality report.
+/// See https://docs.gitlab.com/ee/ci/testing/code_quality.html#implement-a-custom-tool
+#[derive(Serialize)]
+pub(crate) struct GitLabDiagnostic {
+    /// A description of the code quality violation.
+    description: String,
+    /// A unique name representing the static analysis check that emitted this issue.
+    check_name: String,
+    /// A unique fingerprint to identify the code quality violation. For example, an MD5 hash.
+    fingerprint: String,
+    /// A severity string (can be info, minor, major, critical, or blocker).
+    severity: GitLabSeverity,
+    /// The location where the code quality violation occurred.
+    location: Location,
+}
```

Comment:
I really think you should reconsider the approach, and use the same logic we use for GitHub: https://github.com/biomejs/biome/blob/main/crates/biome_diagnostics/src/display_github.rs

We have two benefits:
- we don't allocate new information (strings in this case) for information **we already have**
- we use the `biome_console::fmt::Display` trait, which is correct approach